api/src/test/java/org/apache/iceberg/TestPartitionSpecValidation.java

```diff
@@ -38,6 +38,9 @@
   public void testMultipleTimestampPartitions() {
     AssertHelpers.assertThrows("Should not allow year(ts) and year(ts)",
         IllegalArgumentException.class, "Cannot use partition name more than once",
+        () -> PartitionSpec.builderFor(SCHEMA).year("ts", "year").year("another_ts", "year").build());
```

Comment:
The context for these checks is incorrect. The problem is not a duplicate transform for a field, it is that the name is reused.

Also, these should not be in this test case for multiple timestamp partitions. These should be in a test case for duplicate names.

Reply:
I think this change is the side effect if the duplicate partition detection takes precedence over duplicate name detection as discussed in https://github.com/apache/iceberg/pull/922#discussion_r441199360.

Reply:
The context, `Should not allow year(ts) and year(ts)` isn't correct because the call uses two different source columns, `ts` and `another_ts`. It should be `Should not allow partition fields with the same name`.

And since this is passing in the partition name, you can change it to avoid hitting the wrong error case. You could move this to a new test for name collisions, and update this to avoid the name collision by removing the explicit partition field name.